src/TreeSupport.h

```diff
@@ -7,10 +7,92 @@
 #include <forward_list>
 #include <unordered_set>
 
+#include <memory>
+#include <vector>
+#include <utility>
+#include <unordered_map>
+
 #include "sliceDataStorage.h"
 
+
 namespace cura
 {
+/*!
+ * \brief Lazily generates tree guidance volumes.
+ *
+ * NOTE: This class is not currently thread-safe and should not be accessed in OpenMP blocks
+ */
+class ModelVolumes
```

Comment:
why is this functionality in a separate class?

Reply:
It allows the details of generating the various model volumes to be abstracted out of the tree support generation code. This both helps the TreeSupport code become cleaner (as it doesnt have the collision volume generation mixed in) and makes the coding for the lazy generation much simpler.

Theres not reason it _couldn't_ be implemented inside the TreeSupport object, but i'd argue theres no good reason it _should_.

Reply:
Okay